ui/app/src/Views/SilenceForm/Types.elm

```diff
@@ -45,46 +63,53 @@ toSilence { id, createdBy, comment, startsAt, endsAt, matchers } =
             , status = nullSilenceStatus
             }
         )
-        (timeFromString startsAt)
-        (timeFromString endsAt)
+        (timeFromString form.startsAt.value)
+        (timeFromString form.endsAt.value)
         |> Result.fromMaybe "Wrong datetime format"
 
 
 fromSilence : Silence -> SilenceForm
 fromSilence { id, createdBy, comment, startsAt, endsAt, matchers } =
     { id = id
-    , createdBy = createdBy
-    , comment = comment
-    , startsAt = timeToString startsAt
-    , endsAt = timeToString endsAt
-    , duration = durationFormat (endsAt - startsAt)
-    , matchers = matchers
+    , createdBy = validateCreatedBy createdBy
+    , comment = validateComment comment
+    , startsAt = validateStartsAt (timeToString startsAt)
+    , endsAt = validateEndsAt (timeToString endsAt)
+    , duration = validateDuration (durationFormat (endsAt - startsAt))
+    , matchers = List.map validateMatcher matchers
     }
 
 
 empty : SilenceForm
 empty =
     { id = ""
-    , createdBy = ""
-    , comment = ""
-    , startsAt = ""
-    , endsAt = ""
-    , duration = ""
+    , createdBy = validateCreatedBy ""
+    , comment = validateComment ""
+    , startsAt = validateStartsAt ""
+    , endsAt = validateEndsAt ""
+    , duration = validateDuration ""
     , matchers = []
     }
 
 
 fromMatchersAndTime : List Matcher -> Time -> SilenceForm
-fromMatchersAndTime matchers now =
+fromMatchersAndTime maybeEmptyMatchers now =
```

Comment:
I would vote for not using `maybe` in the variable name, as that signals the `Maybe` type to me when first reading it